network/openvswitch_bridge.py

```diff
@@ -80,10 +95,15 @@ class OVSBridge(object):
     def __init__(self, module):
         self.module = module
         self.bridge = module.params['bridge']
+        self.parent = module.params['parent']
+        self.vlan = module.params['vlan']
         self.state = module.params['state']
         self.timeout = module.params['timeout']
         self.fail_mode = module.params['fail_mode']
 
+        if self.parent and not self.vlan:
+            self.module.fail_json(msg='VLAN id must be set when parent is defined')
+
```

Comment:
It would be good to have an additional check that the VLAN ID is valid:

```python
        if self.vlan < 0 or self.vlan > 4095:
            self.module.fail_json(msg='Invalid VLAN ID (must be between 0 and 4095)')
```